shared/pub/metadata.js

```diff
@@ -0,0 +1,54 @@
+export const getPDFDownloads = (pub) => {
+	const downloads = pub.downloads;
+	const exports = pub.activeBranch.exports;
+	if (downloads && downloads.length > 0) {
+		return downloads.filter((download) => {
+			if (download.url.substring(download.url.length - 4) === '.pdf') {
+				return download;
+			}
+			return false;
+		});
+	}
+	if (exports && exports.length > 0) {
+		return exports.filter((exportFile) => {
+			if (exportFile.format === 'pdf') {
+				return exportFile;
+			}
+			return false;
+		});
+	}
+	return false;
+};
+
+export const getTextAbstract = (content) => {
+	let abstract = '';
+	if (!content) {
+		return abstract;
+	}
+	if (
+		content.content[0].type === 'heading' &&
+		content.content[0].attrs.level === 1 &&
+		content.content[0].attrs.id === 'abstract'
```

Comment:
This is probably fine, but I think it might be more robust to check the h1 content than its `attrs.id`, which we might repurpose for intra-document linking in the future.